cql3/expr/expression.cc

```diff
@@ -1352,5 +1357,224 @@ std::vector<expression> extract_single_column_restrictions_for_column(const expr
 }
 
 
+constant::constant(cql3::raw_value val, data_type typ)
+    : value(std::move(val)), type(std::move(typ)) {
+}
+
+constant constant::make_null(data_type val_type) {
+    return constant(cql3::raw_value::make_null(), std::move(val_type));
+}
+
+constant constant::make_unset_value(data_type val_type) {
+    return constant(cql3::raw_value::make_unset_value(), std::move(val_type));
+}
+
+constant constant::make_bool(bool bool_val) {
+    return constant(raw_value::make_value(boolean_type->decompose(bool_val)), boolean_type);
+}
+
+bool constant::is_null() const {
+    return value.is_null();
+}
+
+bool constant::is_unset_value() const {
+    return value.is_unset_value();
+}
+
+bool constant::has_empty_value_bytes() const {
+    if (is_null_or_unset()) {
+        return false;
+    }
+
+    return value.to_view().size_bytes() == 0;
+}
+
+bool constant::is_null_or_unset() const {
+    return is_null() || is_unset_value();
+}
+
+std::optional<bool> get_bool_value(const constant& constant_val) {
+    if (constant_val.type->get_kind() != abstract_type::kind::boolean) {
+        return std::nullopt;
+    }
+
+    if (constant_val.is_null_or_unset()) {
+        return std::nullopt;
+    }
+
+    if (constant_val.has_empty_value_bytes()) {
```

Comment:
I'm not sure if it is possible to encounter a value like this in the wild, but some other types can be `empty` - not `null` or `unset`, but they have 0 bytes. I think it is good to protect against such cases. Maybe some driver allows to insert empty values?
We still need the `null_or_unset` check, because an `empty` value is not `null` or `unset`